app/controllers/feedback_controller.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class FeedbackController < ApplicationController
+  before_action :authenticate_user!
+  before_action :feature_enabled?
+  before_action :canny_consent_given?, only: %w[features bugs]
+
+  def consent
+    redirect_to feedback_bugs_path if current_user.has_role? :canny_consent
+
+    return unless params[:consent] == "true"
+
+    current_user.add_role :canny_consent
+    redirect_to feedback_bugs_path
```

Comment:
so if a user clicked on `Feature requests` and they didn't consent yet they'll end up at the `Bugs` page?

Reply:
Yes.

I'm a bit conflicted on _how much_ of an issue this ends up being, since the tabs at the top are relatively large.

![image](https://user-images.githubusercontent.com/1774242/150654086-bb84f688-c2a0-40e9-ba7f-dc0bc7f55939.png)

Otherwise it shouldn't be that hard to implement that.

Reply:
Will open an issue after this is merged because a simple gate to prevent external URL redirects is only added in Rails 7 🤦